Aliucord/src/main/java/com/aliucord/api/ButtonsAPI.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021 Juby210 & Vendicated
+ * Licensed under the Open Software License version 3.0
+ */
+
+package com.aliucord.api
+
+import android.content.Context
+import androidx.fragment.app.FragmentManager
+
+import com.aliucord.Logger
+import com.aliucord.utils.ReflectUtils
+import com.aliucord.coreplugins.ButtonsAPI
+
+import com.discord.api.botuikit.*
+import com.discord.models.message.Message
+import com.discord.stores.StoreStream
+
+import java.util.*
+import java.lang.reflect.*
+import kotlin.collections.lastOrNull
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KProperty
+
+/**
+ * Adds methods for creating button components
+ */
+@Suppress("UNCHECKED_CAST")
+object ButtonsAPI {
+
+    /**
+     * Stores data about a button
+     */
+    data class ButtonData(val label: String, val style: ButtonStyle, val onPress: (Message, FragmentManager) -> Unit)
```

Comment:
onPress should pass an android context to callback, maybe you can pass activity instead of fragmentmanager? it is both context and access to fragmentManager